source/Main.cpp

```diff
@@ -34,23 +36,26 @@ int main(int argc, char* argv[]) {
 
   namespace program_options = boost::program_options;
   program_options::options_description options;
-  options.add_options()("help,h", "Show help dialog.");
+  options.add_options()
+  ("help,h", "Show help dialog.")
+  ("config,c", program_options::value<std::string>()->required(), "Path to the JSON configuration file.");
 
   auto tool = marianatrench::MarianaTrench();
   tool.add_options(options);
 
   try {
     program_options::variables_map variables;
-    program_options::store(
-        program_options::command_line_parser(argc, argv).options(options).run(),
-        variables);
-
+    program_options::store(program_options::parse_command_line(argc, argv, options), variables);
     if (variables.count("help")) {
       std::cerr << options;
       return 0;
     }
-
-    program_options::notify(variables);
+    if (!variables.count("config")) {
+      std::cerr << "No JSON configuration file provided.\n";
+      std::cerr << "Usage: " << argv[0] << " --config <json_config_file>\n";
+      return ExitCode::invalid_argument_error("No JSON configuration file provided.");
+    }
+    
```

Comment:
Is this necessary? I think boost should already error when calling `program_options::parse_command_line` since you marked `config` as required.

Reply:
It is necessary, I tried to remove the `--config` from the Python shim and it didn't give an error.